development/db-schema.md

```diff
@@ -1,158 +1,150 @@
+XML forms are stored in the database and have:
+ - An `_id` of `form:<formname>`
+ - The `type` of `form`
+ - The actual XML Xforms definition attached
 
-Example of record that gets created from an SMS message, after a message is submitted and form is parsed:
+XML forms are defined as XForm XML files
 
-```json
-{
-  "_id": "3d9c7c94",
-  "_rev": "5-f49f0dbc",
-  "type": "data_record",
-  "from": "+9779988777766",
-  "form": "P",
-  "errors": [],
-  "tasks": [],
-  "reported_date": 1484308744000,
-  "sms_message": {
-    "secret": "",
-    "from": "+9779988777766",
-    "message": "P 15 लछुता",
-    "sent_timestamp": "1484308744000",
-    "message_id": "6de6bb42",
-    "type": "sms_message",
-    "form": "P",
-    "locale": "ne"
-  },
-  "fields": {
-    "last_menstrual_period": 15,
-    "patient_name": "लछुता",
-  }
-}
-```
-
-For SMS reports, after the report is initially saved other processes will come and modify it, typically the next thing that happens is medic-sentinel will populate the `place` and `contact` properties.  This requires a database query so is done as a separate process from the initial creation.
-SMS form definitions are part of application settings, on the design doc; so it can be used to parse data during initial record creation.
-
-For XML reports, the actual xml data is also saved in attachments to the doc. (*Legacy: The actual xml data is saved in doc, in `content` field.*)
-
-Data_record has the internalId of the form it comes from, in `form`.
-Data_record has a copy of the `type: person` doc that submitted it, in `contact` (SMS-generated reports may not have a contact)
-If sent by SMS, it can also have details of the SMS message, and the `scheduled_tasks` triggered by the form submission.
+## Users
 
-## Meta
+Users represent credentials and roles / permissions for accessing the application. This can either be:
+ - people who can log into the application, such as CHWs or Nurses
+ - or credentials granting external software restricted permissions to perform certain tasks, such as allowing an external service permission to write reports via the api.
 
-`type: meta` : for keeping track of which migrations have been run by `medic-api`. See migrations.js.
+User records have at least:
+ - An `_id` of `org.couchdb.user:<username>`
+ - A `name` which is the same as `<username>` above
+ - A `roles` array
 
-## Feedback
+There are two slightly different copies of this record stored.
 
-`type: feedback` : created when used clicks “Report bug”
+The `_users` database additionally:
+ - The `type` of `user`
+ - The password hash and associated data
+ - and is a location that only administrative users can write to, and so is authoritive when it comes to roles and the like.
+ - It is also what CouchDB usess for authentication
```

Comment:
These last two points should be just written as sentences in this section rather than bullet points as they don't fit with the first two.